dvc/repo/status.py

```diff
@@ -113,4 +115,15 @@ def status(
             remote=remote,
             all_tags=all_tags,
         )
+
+    ignored = list(
+        compress(
+            ["all_branches", "all_tags", "jobs", "remote", "cloud"],
+            [all_branches, all_tags, jobs, remote, cloud],
+        )
+    )
+    if ignored:
+        msg = "the following options are meaningless for local status: {}"
+        raise DvcException(msg.format(", ".join(ignored)))
```

Comment:
The CLI user will see "the following options are meaningless for local status: all_branches, ..." which might be confusing, as those don't match with CLI flags. Also you have `cloud` there, which tells that this is a non-local status :slightly_smiling_face: Let's do something like https://github.com/iterative/dvc/blob/master/dvc/command/run.py#L28 instead, where we catch this in CLI.

Reply:
But it should be handled in repo method as well.

Reply:
@Suor Ideally yes, but in other places we either only catch that in CLI and ignore in API or we don't hesitate to use CLI flags in exceptions that we use in API. Let's either stick to one of those (I think the former is a quick and dirty way, but still reasonable) for now and/or we could think of a general solution that would catch that in API and CLI and present a relevant message.